python/cudf/cudf/_lib/groupby.pyx

```diff
@@ -73,6 +73,21 @@ _DECIMAL_AGGS = {
 ctypedef const scalar constscalar
 
 
+cdef get_valid_aggregation(object dtype):
```

Comment:
Rather than a cdef function, I wonder if it would be worth using a `functools.singledispatch` here. We're not benefiting from the `cdef` because everything is a pure Python type, and the `isinstance` cascade isn't free, especially if groupby is run many times.

Reply:
Good suggestion, singledispatch is a good alternative so I'll use that